libraries/AP_RPM/AP_RPM.cpp

```diff
@@ -110,21 +110,22 @@ void AP_RPM::init(void)
         return;
     }
     for (uint8_t i=0; i<RPM_MAX_INSTANCES; i++) {
+#if CONFIG_HAL_BOARD != HAL_BOARD_SITL
         uint8_t type = _type[i];
-
         if (type == RPM_TYPE_PWM) {
             // PWM option same as PIN option, for upgrade
             type = RPM_TYPE_PIN;
         }
         if (type == RPM_TYPE_PIN) {
             drivers[i] = new AP_RPM_Pin(*this, i, state[i]);
         }
-#if EFI_ENABLED
+
+    #if EFI_ENABLED
```

Comment:
this breaks EFI in SITL

Reply:
I have now moved the EFI driver instantiation outside of the check for HAL_BOARD_SITL.

Reply:
that still doesn't look right, as it will allocate AP_RPM_SITL, then overwrite it with AP_RPM_EFI object?

Reply:
Ah ok, sorry didn't see that.  I have now changed it again.  I think it is correct now.